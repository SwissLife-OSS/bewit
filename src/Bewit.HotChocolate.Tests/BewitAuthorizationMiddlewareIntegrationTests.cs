using System;
using System.Threading.Tasks;
using Bewit.Generation;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using Squadron;
using Xunit;

namespace Bewit.HotChocolate.Tests
{
    public class BewitAuthorizationMiddlewareIntegrationTests
        : IClassFixture<MongoResource>
    {
        private readonly MongoResource _mongoResource;

        public BewitAuthorizationMiddlewareIntegrationTests(MongoResource mongoResource)
        {
            _mongoResource = mongoResource;
        }

        [Fact]
        public async Task Query_WhenAuthorize_Success()
        {
            // arrange
            var serviceProvider = TestHelpers.CreateServiceProvider(_mongoResource);
            var payload = new CustomPayload {Email = "foo@bar.gmail.com"};
            var token = await TestHelpers.CreateToken(serviceProvider, payload);
            var schema = TestHelpers.CreateSchema(serviceProvider);

            // act
            var result = await TestHelpers.ExecuteQuery(schema, token);

            // assert
            var bewitContext = serviceProvider.GetService<IBewitContext>();
            var customPayload = await bewitContext.GetAsync<CustomPayload>();
            new { QueryResult = result, BewitContext = customPayload }.MatchSnapshot();
        }
    }
}