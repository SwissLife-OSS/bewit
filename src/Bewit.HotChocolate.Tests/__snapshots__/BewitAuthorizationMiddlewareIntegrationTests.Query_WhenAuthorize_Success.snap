{
  "QueryResult": {
    "Data": {
      "foo": "bar"
    },
    "Errors": null,
    "Extensions": null,
    "ContextData": null
  },
  "BewitContext": {
    "Email": "foo@bar.gmail.com"
  }
}
